mmdet/models/layers/transformer/dab_detr_transformer.py

```diff
@@ -0,0 +1,667 @@
+                **kwargs)
+            # iter update
+            if reg_branches is not None:
+                tmp = reg_branches(output)
+                tmp[..., :self.query_dim] += inverse_sigmoid(reference)
+                new_reference = tmp[..., :self.query_dim].sigmoid()
+                if layer_id != self.num_layers - 1:
+                    ref.append(new_reference)
+                reference = new_reference.detach()  # no grad_fn
+            if self.return_intermediate:
+                if self.post_norm is not None:
+                    intermediate.append(self.post_norm(output))
+                else:
+                    intermediate.append(output)
+
+        if self.post_norm is not None:
+            output = self.post_norm(output)
+
+        if reg_branches is not None and self.return_intermediate:
+            return [
+                torch.stack(intermediate).transpose(1, 2),
+                torch.stack(ref).transpose(1, 2),
+            ]
+        elif reg_branches is None and self.return_intermediate:
+            return [
+                torch.stack(intermediate).transpose(
+                    1, 2),  # return_intermediate is True
+                reference.unsqueeze(0).transpose(1, 2)  # reg_branches is None
+            ]
+        elif reg_branches is None and not self.return_intermediate:
+            return [
+                output.unsqueeze(0).transpose(
+                    1, 2),  # return_intermediate is False
+                reference.unsqueeze(0).transpose(1, 2)  # reg_branches is None
+            ]
+        else:
+            return [
+                output.unsqueeze(0).transpose(
+                    1, 2),  # return_intermediate is False
+                torch.stack(ref).transpose(1, 2)  # reg_branches is not None
+            ]
+
+
+class DabDetrTransformerEncoder(DetrTransformerEncoder):
+    """Encoder of DAB-DETR."""
+
+    def _init_layers(self):
+        """Initialize encoder layers."""
+        self.layers = ModuleList([
+            DetrTransformerEncoderLayer(**self.layer_cfg)
+            for _ in range(self.num_layers)
+        ])
+        embed_dims = self.layers[0].embed_dims
+        self.embed_dims = embed_dims
+        self.query_scale = MLP(embed_dims, embed_dims, embed_dims, 2)
+
+    def forward(self,
+                query,
+                query_pos=None,
+                query_key_padding_mask=None,
+                **kwargs):
+        """Forward function of encoder.
+
+        Args:
+            query (Tensor): Input queries of encoder, has shape
+                (num_query, bs, dim).
+            query_pos (Tensor): The positional embeddings of the queries, has
+                shape (num_feat, bs, dim).
+            query_key_padding_mask (Tensor): ByteTensor, the key padding mask
+                of the queries, has shape (num_feat, bs).
+
+        Returns:
+            Tensor: With shape (bs, num_query, dim) if `batch_first` is `True`,
```

Comment:
Check the batch_first

Reply:
remove all batch first related comments and make batch first as the only behavior